src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs

```diff
@@ -0,0 +1,350 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DotNet.Status.Web.Options;
+using DotNet.Status.Web.Controllers;
+using Microsoft.DotNet.GitHub.Authentication;
+using Microsoft.DotNet.Internal.AzureDevOps;
+using Microsoft.DotNet.Internal.Testing.Utility;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using FluentAssertions;
+using Moq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using Octokit;
+
+namespace DotNet.Status.Web.Tests
+{
+
+    [TestFixture]
+    public class AzurePipelinesControllerTests
+    {
+        public static string TestTeamsWebHookUri = "https://example.teams/webhook/sha";
+        public static string WatchedTeam = "test-user/watched-team";
+        public static string IgnoredRepo = "test-user/ignored";
+
+        [Test]
+        public async Task BuildCompleteBuildHasNoTags()
+        {
+            var buildEvent = new AzurePipelinesController.AzureDevOpsEvent<AzurePipelinesController.AzureDevOpsMinimalBuildResource>(){
+                Resource = new AzurePipelinesController.AzureDevOpsMinimalBuildResource() {
+                    Id = 123456,
+                    Url = "test-build-url"
+                },
+                ResourceContainers = new AzurePipelinesController.AzureDevOpsResourceContainers(){
+                    Collection = new AzurePipelinesController.HasId() {
+                        Id = "test-collection-id"
+                    },
+                    Account = new AzurePipelinesController.HasId() {
+                        Id = "test-account-id"
+                    },
+                    Project = new AzurePipelinesController.HasId() {
+                        Id = "test-project-id"
+                    }
+                }
+            };
+
+            var build = new JObject
```

Comment:
We don't use all these values, do we?  We should only send the bits we care about. :-)

Reply:
These are actually all used somewhere in the code path.